voicemail-api/javascript/index.html

```diff
@@ -0,0 +1,231 @@
+            'sortOrder': "asc" // String | Sort order
+        };
+
+        routingApi.getRoutingQueuesMe(opts)
+            .then((queueData) => {
+                console.log(`getRoutingQueuesMe success! data: ${JSON.stringify(queueData, null, 2)}`);
+                let queueDetails = queueData.entities;
+                queueDetails.forEach(data => createQueueList(data.id, data.name));
+            })
+            .catch((err) => {
+                console.log('There was a failure calling getRoutingQueuesMe');
+                console.error(err);
+            });
+    }
+
+    // Get all voicemail details associated with current user 
+    function getMeVoicemails() {
+        let opts = {
+            'pageSize': 100, // Number | Page size
+            'pageNumber': 1 // Number | Page number
+        };
+
+        voicemailApi.getVoicemailMeMessages(opts)
+            .then((voicemailData) => {
+                console.log(`getVoicemailMeMessages success! data: ${JSON.stringify(voicemailData, null, 2)}`);
+                getMessageId(voicemailData);
+            })
+            .catch((err) => {
+                console.log('There was a failure calling getVoicemailMeMessages');
+                console.error(err);
+            });
+    }
+    
+    // Get voicemail details for the selected queue
+    function getQueueVoicemails(id) {
+        let queueId = id[id.selectedIndex].id;; // String | Queue ID
+        let opts = {
+            'pageSize': 100, // Number | Page size
+            'pageNumber': 1 // Number | Page number
+        };
+
+        voicemailApi.getVoicemailQueueMessages(queueId, opts)
+            .then((queueData) => {
+                console.log(`getVoicemailQueueMessages success! data: ${JSON.stringify(queueData, null, 2)}`);
+                voicemailDetails = queueData.entities;
+                voicemailDetails.forEach(data => getVoiceMailMedia(data.id, data.createdDate, data.callerAddress,
+                    data.audioRecordingDurationSeconds));
+            })
+            .catch((err) => {
+                console.log('There was a failure calling getVoicemailQueueMessages');
+                console.error(err);
+            });
+    }
+
+    // Get ID needed to retreive voicemail as well as other details that needs to be passed on and displayed
+    function getMessageId(voicemailData) {
+        voicemailDetails = voicemailData.entities;
+        voicemailDetails.forEach(data => getVoiceMailMedia(data.id, data.createdDate, data.callerAddress, data
+            .audioRecordingDurationSeconds));
+    }
+
+    // Using the messageId, get the voicemail media and download it as your desired media format
+    function getVoiceMailMedia(id, date, ani, duration) {
+        console.log("getVoiceMailMedia function " + id, date, ani, duration);
+        let messageId = id; // String | Message ID
+        let opts = {
+            'formatId': "WAV" // String | The desired media format.

```

Comment:
You might want to add in the comment again the list of other format types they can ask to retrieve

Reply:
done adding other format types